Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,424 @@
+                                                }
+                                                else
+                                                {
+                                                    WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(applicationModel);
+                                                }
+                                            }
+                                        }
+                                        catch (Exception ex)
+                                        {
+                                            Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        bf.ConversionStatus = eConversionStatus.Finish;
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method will create the new api model action
+        /// </summary>
+        /// <param name="applicationModel"></param>
+        /// <param name="act"></param>
+        /// <returns></returns>
+        private ActWebAPIModel GetNewAPIModelAction(Guid applicationModelGuid, Act act)
+        {
+            ActWebAPIModel actModel = new ActWebAPIModel();
+            actModel.ItemName = string.Format("New - {0}", act.ItemName);
+            actModel.Description = string.Format("New - {0}", act.Description);
+            actModel.APImodelGUID = applicationModelGuid;
+            actModel.ReturnValues = act.ActReturnValues;
+            actModel.MaxNumberOfRetries = act.MaxNumberOfRetries;
+            actModel.Active = true;
+            return actModel;
+        }
+
+        /// <summary>
+        /// This method is used to check if the app parameter is already added in the list if not then it will add the parameter
+        /// </summary>
+        /// <param name="aPIModel"></param>
+        /// <param name="act"></param>
+        /// <param name="actApiModel"></param>
+        private void AddAppParameters(ApplicationAPIModel aPIModel, Act act, ActWebAPIModel actApiModel)
+        {
+            try
+            {
+                ObservableList<AppModelParameter> lstParameters = null;
+                ObservableList<ActInputValue> actInputs = ((ActWebAPIBase)act).DynamicElements;
+                if (actInputs == null || actInputs.Count == 0)
+                {
+                    string requestBody = string.Empty;
+                    if (aPIModel.RequestBodyType == ApplicationAPIUtils.eRequestBodyType.TemplateFile)
+                    {
+                        int indexBF = aPIModel.ContainingFolder.IndexOf("BusinessFlows");
```

Comment:
how BusinessFlows path is releveant for getting the path of template body? only need to replace ~ with Solution path and we already have generic function which does is in SolutionRepository

Reply:
Changed